internal/hardware/hardware.go

```diff
@@ -36,6 +62,22 @@ func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
 		}
 		hardwareInfo.Interfaces = append(hardwareInfo.Interfaces, newInterface)
 	}
+}
 
-	return &hardwareInfo, nil
+func (s *Hardware) Init() {
+	if s.Dependencies == nil {
```

Comment:
I would take xtra care here, this will be called from heartbeat(goroutine) and registration (can be another goroutine) some race conditions can happen here. Atomic or mutex is recommended.

Reply:
You mean that the function calling Init could be called simultaneously by different goroutines? 
Because, actually, this Init may be private and should only be called by internal function, I missed this one
I added the dependencies field mainly to allow testing the retrieval of hardware info 
IMO, using mutex could consume too much resource compared to the service provided